go/interpreter/lfvm/instruction_logger_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package lfvm
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+)
+
+func TestInterpreter_Logger_ExecutesCodeAndLogs(t *testing.T) {
+
+	tests := map[string]struct {
+		code []Instruction
+		want string
+	}{
+		"empty": {},
+		"stop": {
+			code: []Instruction{{STOP, 0}},
+			want: "STOP, 3, -empty-\n",
+		},
+		"multiple codes": {
+			code: []Instruction{{PUSH4, 0}, {DATA, 1}, {STOP, 0}},
+			want: "PUSH4, 3, -empty-\nSTOP, 0, 1\n",
+		},
+		"out of gas": {
+			code: []Instruction{
+				{PUSH1, 0},
+				{PUSH1, 64},
+				{MSTORE8, 0},
+				{STOP, 0},
+			},
+			want: "PUSH1, 3, -empty-\nPUSH1, 0, 0\n",
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+
+			// Get tosca.Parameters
+			params := tosca.Parameters{
+				Gas:  3,
+				Code: []byte{byte(STOP), 0},
```

Comment:
This code here should be irrelevant and be removed because it is inconsistent with the actual code executed.

It seems like it is only needed because of this check here:
https://github.com/Fantom-foundation/Tosca/blob/3ba757ff83347886881383df638ca2143aed6509/go/interpreter/lfvm/interpreter.go#L89

This check should be `if len(code) == 0`, not `if len(params.Code) == 0`. Feel free to update this as part of this PR.